pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -431,27 +456,37 @@ func TestAddPodWillReplaceAssumed(t *testing.T) {
 	}}
 
 	for i, tt := range tests {
-		cache := newSchedulerCache(ttl, time.Second, nil)
-		for _, podToAssume := range tt.podsToAssume {
-			if err := assumeAndFinishBinding(cache, podToAssume, now); err != nil {
-				t.Fatalf("assumePod failed: %v", err)
-			}
-		}
-		for _, podToAdd := range tt.podsToAdd {
-			if err := cache.AddPod(podToAdd); err != nil {
-				t.Fatalf("AddPod failed: %v", err)
-			}
-		}
-		for _, podToUpdate := range tt.podsToUpdate {
-			if err := cache.UpdatePod(podToUpdate[0], podToUpdate[1]); err != nil {
-				t.Fatalf("UpdatePod failed: %v", err)
-			}
-		}
-		for nodeName, expected := range tt.wNodeInfo {
-			t.Log(nodeName)
-			n := cache.nodes[nodeName]
-			deepEqualWithoutGeneration(t, i, n, expected)
-		}
+		t.Run(fmt.Sprintf("%v", i), func(t *testing.T) {
+			cache := newSchedulerCache(ttl, time.Second, nil)
+			t.Run("assumePod", func(t *testing.T) {
+				for _, podToAssume := range tt.podsToAssume {
+					if err := assumeAndFinishBinding(cache, podToAssume, now); err != nil {
+						t.Fatalf("failed: %v", err)
+					}
+				}
+			})
+			t.Run("AddPod", func(t *testing.T) {
+				for _, podToAdd := range tt.podsToAdd {
+					if err := cache.AddPod(podToAdd); err != nil {
+						t.Fatalf("failed: %v", err)
+					}
+				}
+			})
+			t.Run("UpdatePod", func(t *testing.T) {
+				for _, podToUpdate := range tt.podsToUpdate {
+					if err := cache.UpdatePod(podToUpdate[0], podToUpdate[1]); err != nil {
```

Comment:
podToUpdate should have been a struct with "old" and "new" pods, rather than a slice. Probably don't fix it in this PR.